### PR TITLE
Add KMS interpreter and service

### PR DESCRIPTION
**Pull request by marctalbott**
This is still a wip as it may change depending on the automation test I'm currently working on for Sam, but this is what I've got so far

**PR checklist**
- [x] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [x] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [x] Get two thumbsworth of PR review
- [x] Verify all tests go green (CI _and_ coverage tests)
- [x] Squash commits and **merge to develop**
- [x] Delete branch after merge

### CODE REVIEW

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import scala.language.higherKinds
+
+/**
+  * Interpreter for Google KMS (Key Management Service) algebra
+  *
+  * created by mtalbott on 1/18/19
+  */
+
+private[google] class GoogleKmsInterpreter[F[_]: Sync: ContextShift](client: KeyManagementServiceClient, blockingEc: ExecutionContext) extends GoogleKmsService[F] {
+  override def createKeyRing(project: String, location: String, keyRingId: String): F[KeyRing] = {
```
Comment (Qi77Qi) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala]: there's a `GoogleProject` type I think.
I'll make custom type for `location` and `keyRingId`. Stringy types are too error prone

Same comment goes for other methods

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+      _ <- setIamPolicy(project, location, keyRingId, keyId, newPolicy)
+    } yield newPolicy
+  }
+
+  override def removeMemberFromKeyPolicy(project: String, location: String, keyRingId: String, keyId: String, member: String, role: String): F[Policy] = {
+    for {
+      currentIamPolicyOpt <- getIamPolicy(project, location, keyRingId, keyId)
+
+      otherBindings = currentIamPolicyOpt.get.getBindingsList.asScala.toList.filter(binding => !binding.getRole.equals(role))
```
Comment (Qi77Qi) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala]: calling `.get` on `Option` will throw exception. It should be avoided in general. I think for this case, you can use `map` to get inside the `Option`, and at the end do `.getOrElse(List.empty)`

Something like `currentIamPolicyOpt.map(x => x.getBindingsList.asScala.toList.filter(binding => !binding.getRole.equals(role)).getOrElse(List.empty)`

This way you don't need to worry about exceptions.


Also, double check the equality check works the way you expect

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      newBindings = currentIamPolicyOpt.get.getBindingsList.asScala.toList.filter(binding => binding.getRole.equals(role)).map { binding =>
```
Comment (Qi77Qi) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala]: same comment. Don't use `.get` on `Option`, it defeats the purpose of using `Option` in the first place.

File: google/src/test/scala/org/broadinstitute/dsde/workbench/google/mock/MockGoogleKmsInterpreter.scala [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import com.google.iam.v1.Policy
+import org.broadinstitute.dsde.workbench.google.GoogleKmsService
+
+/**
+  * Mock Google Key Management Service Interpreter
+  *
+  * created by mtalbott 1/28/19
+  */
+
+object MockGoogleKmsInterpreter extends GoogleKmsService[IO] {
```
Comment (Qi77Qi) [inline google/src/test/scala/org/broadinstitute/dsde/workbench/google/mock/MockGoogleKmsInterpreter.scala]: minor, I kinda prefer `Fakexxx`. It's the same naming as some emulators, and it's easy to know that it's new google code as opposed to older google module.

**Comment by Qi77Qi** [comment]
👍  overall looks really good...left a few small comments

File: google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+  override def addMemberToKeyPolicy(project: GoogleProject, location: Location, keyRingId: KeyRingId, keyId: KeyId, member: String, role: String): F[Policy] = {
+    for {
+      currentIamPolicyOpt <- getIamPolicy(project, location, keyRingId, keyId)
+
+      newBinding = Binding.newBuilder()
+        .setRole(role)
+        .addMembers(member)
+        .build()
+      newPolicy = Policy.newBuilder(currentIamPolicyOpt.get)
```
Comment (dvoet) [inline google/src/main/scala/org/broadinstitute/dsde/workbench/google/GoogleKmsInterpreter.scala]: .get is bad practice... how can you handle None?

**Review by dvoet — CHANGES_REQUESTED** [review]
1 little thing

**Review by dvoet — APPROVED** [review]

**Review by Qi77Qi — APPROVED** [review]
thought I approved it a while back..Maybe the approval is gone since you rebased?

**Comment by Qi77Qi** [comment]
nvm..I was thinking of another PR

**Pull request merged**